gt/main.c

```diff
@@ -1697,14 +1695,37 @@ gt_stage2(void *arg)
 }
 
 int
-run_gt(struct net_config *net_conf, struct gt_config *gt_conf)
+run_gt(struct net_config *net_conf, struct gt_config *gt_conf,
+	const char *lua_base_directory, const char *lua_policy_file)
 {
 	int ret, i;
 
-	if (net_conf == NULL || gt_conf == NULL) {
+	if (net_conf == NULL || gt_conf == NULL ||
+			lua_base_directory == NULL ||
+			lua_policy_file == NULL) {
+		ret = -1;
+		goto out;
+	}
+
+	gt_conf->lua_base_directory = rte_malloc(
+		"lua_base_directory", strlen(lua_base_directory) + 1, 0);
+	if (gt_conf->lua_base_directory == NULL) {
+		RTE_LOG(ERR, MALLOC, "%s: Out of memory for Lua policy base directory name\n",
+			__func__);
 		ret = -1;
 		goto out;
 	}
+	strcpy(gt_conf->lua_base_directory, lua_base_directory);
+
+	gt_conf->lua_policy_file = rte_malloc(
+		"lua_policy_file", strlen(lua_policy_file) + 1, 0);
+	if (gt_conf->lua_policy_file == NULL) {
+		RTE_LOG(ERR, MALLOC, "%s: Out of memory for Grantor configuration file name\n",
+			__func__);
+		ret = -1;
+		goto policy_dir;
+	}
+	strcpy(gt_conf->lua_policy_file, lua_policy_file);
 
 	if (!(gt_conf->gt_max_pkt_burst > 0)) {
 		ret = -1;
```

Comment:
I can't comment on the `goto out`'s below, but shouldn't they be replaced with `goto gt_config_file`?